pkg/registry/service/rest.go

```diff
@@ -157,7 +158,7 @@ func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		}
 	}
 
-	if featuregate.DefaultFeatureGate.ExternalTrafficLocalOnly() && shouldCheckOrAssignHealthCheckNodePort(service) {
+	if shouldCheckOrAssignHealthCheckNodePort(service) {
```

Comment:
you have a comment below that I don't understand "Insert the newly allocated health check port as an annotation (plan of record for Alpha)", can you clarify it.